### ISSUE TITLE
Installation of Cobigen Eclipse Integration broken

### ISSUE THREAD

**Issue by Jan-PeterRichterCG** [devonfw/ide#330]
[Note: this Bug is very closely related to https://github.com/devonfw/devonfw-guide/issues/59#issue-544011687 it may even be more of a documentation bug than a ide bug - although the root cause of trouble is the missing supply of plugins unter https://bintray.com/devonfw/cobigen.p2 which is a ide bug.]

### Expected behavior

As a devonfw user, I want to install CobiGen (including Templates) without research so that I can use CobiGen within minutes

### Actual behavior
Installation and documentation is broken in several aspects.

["Chapter 54" in the following lines is devonfw-guide/tools-cobigen.wiki/cobigen-eclipse_installation.asciidoc]

- Chapter 54 of the devonfw guide says: "Remark: CobiGen is preinstalled in the devonfw/devon-ide". At the time of writing this bug report, the out-of-the-box ide is not available (though it is announced to be available soon ==> for a bug"fix", please check whether the Cobigen installation is 100% operational in the out-of-the-box ide. [Is there an automated test suite for the ide?])
- Chapter 54 says: "54.1.1. Preconditions / • Eclipse 4.x / • Java 7 Runtime (for starting eclipse with CobiGen). This is independent from the target version of your developed code." I can hardly believe this information is up to date. If Cobigen really requires Java 7 (and is incompatible with later Java versions) this is a bug of Cobigen in itself ==> please update documentation to accurately reflect the current preconditions.
- Chapter 54 says: "Open CobiGen’s update site /  Insert the update site of your interest into the filed Work with and press Add … / ◦ Stable releases: http://de-mucevolve02/files/cobigen/updatesite/stable/ / ◦ Beta/RC releases: http://de-mucevolve02/files/cobigen/updatesite/experimental/" The URIs are outdated. They sould point to https://bintray.com/devonfw/cobigen.p2. (see #84, https://github.com/devonfw/ide/issues/84#issuecomment-515944453)
- https://bintray.com/devonfw/cobigen.p2 still does not contain the plugin so that they can be downloaded
- Chapter 54 says: "Follow the installation wizard
Select CobiGen Eclipse Plug-in → Next → Next → accept the license → Finish → OK → Yes" while the screenshot above these lines show three tickmarks: "Capgemini Development Tools / CobiGen - Eclipse-Plugin", "Maven osgi bundles / Cobigen - FreeMarker Template Engine", and "Maven osgi bundles / Cobigen - Java Plugin". Several other plugins like "Maven osgi bundles / CobiGen - Open API Plugin" are not ticked ==> please explicitly document what each of these plugins does and what has (should) be installed. Also, please describe the naming conventions and versioning rules: in http://de-mucevolve02/files/cobigen/updatesite/stable/, there are entries like e.g. "Cobigen - Java Plugin 1.7.0.20180126095108" vs "CobiGen - Java Plugin 2.1.0" [which does not obviously correspond to "devon4j 3.1.0" or "devonfw 3.2" or "devonfw-ide 3.2.3". Also, there is "CobiGen - Text Merger 1.2.0" and "Cobigen - Text Merger Plug-in 2.0.0". Apparently, the latter is a newer version of the former, but If you have selected "Show only the latest version of available software" and "Hide items that are already installed" [i.e. the eclipse default configuration] you will see the old "Text Merger" even if you have already installed the newer "Text Merger Plugin" ==> Makee sure that no confusingly named plugins are published in https://bintray.com/devonfw/cobigen.p2
-  Chapter 54 says: "Checkout (clone) your project’s templates folder or use the current templates released with CobiGen (https://github.com/devonfw/tools-cobigen/tree/master/cobigen-templates) and [...]" [You cannot clone directly from this URL since it is a subdirectory.] When you clone from https://github.com/devonfw/tools-cobigen you run into problems with excessively long paths and file names under Windows. ==> please include the hint to ```git config --system core.longpaths true``` [needs to be executed in a cmd started "as Administrator"] as proposed by @markusschuh in https://github.com/devonfw/ide/issues/84#issuecomment-570028235 . Or even better, prepare the devonfw ide in a way that it already includes the hint (may not be feasible unless started "as Administrator" which I would not recommend).
-  Chapter 54 says: "[...] and then choose Import -> General -> Existing Projects into Workspace to import the templates into your workspace." This leads to strange effects since there are currently three *Maven* projects to be imported and all three are called "Cobigen_Templates": templates-devonj4, templates-devon4ng and templates-rf. It appears that it is technically not possible to import all three [because of the identical names], and creates build problems due to missing dependencies and settings. ==> please correct the documentation to "[...] and then choose Import -> Maven -> Existing Maven Projects into Workspace to import the templates into your workspace."
- [Aside: Are the templates-rf open source?]

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

Follow instructions as given in the devonfw guide Chapter 54

### Related/Dependent Issues
#84 

### Affected version:
IDE: Rel 3.2.3; Documentation: As of Dec 2019 (No versioning for the documentation available)

* OS: Windows/Linux/Mac? all
* Browser: Chrome/Firefox/Safari? all

**Comment by maybeec** [comment]
@Jan-PeterRichterCG, as of discussions with legal regarding public distribution of packaged solutions like CobiGen eclipse installation bundles, we needed to take them offline.
Now we are currently in the process to redistribute them by assuring correct licensing texts shipped with the bundles.

**Comment by hohwille** [comment]
@Jan-PeterRichterCG thanks for taking your time to give such feedback.
When it comes to broader observations with many aspects as in this case, it is better to first communicate with the community. We are working in different teams and repositories in devonfw. After all devonfw is a very big project and keeping it maintainable is not always easy.
Hence it is hard for us, to get issues in a repository that affect other repositories that are maintained by other people.
Further as stated by @maybeec your feedback was in the middle of a long ongoing clarification that is now completed. We were fully aware of all the pains with broken download links, etc.
However, this is meanwhile fixed and all done and issues for these problems have already been filed before and are now resolved.
So after all I have to close this issue as invalid as there is IMHO no remaining point in `devonfw-ide` itself that is currently wrong. It might take a few more days for the CobiGen team to get the latest public release complete but from that point on, all should work as promised and as we already provided long time ago with our intermediate internal release before final public legal clarification was completed (see #244, #236, #291, #34, #328, #329, #84).
